### ISSUE TITLE
NET-1710 Fix S1128 FN: Usage of inherited property silence the issue on base class namespace

### ISSUE THREAD

**Issue by sebastien-marichal** [SonarSource/sonar-dotnet#7694]
### Description

Reported in https://community.sonarsource.com/t/unused-using-not-detected-for-a-corner-case/96240.

When using an inherited property while the namespace of the base class is present silence S1128.
This issue seems similar to #4853 

### Repro steps

```cs
namespace ReproS1128.One
{
    public abstract class S1128Base
    {
        public int Id { get; set; }
    }
}
```

```cs
using ReproS1128.One;

namespace ReproS1128.Two
{
    public class S1128Child : S1128Base
    {
    }
}
```

```cs
using ReproS1128.Two;
using ReproS1128.One; // FN - S1128 should raise here as it is an unnecessary using

namespace ReproS1128
{
    public static class S1128
    {
        static void Method()
        {
            var test = new S1128Child();

            test.Id = 42; // When commented out, the issue is raised
        }
    }
}
```

**Comment by pavel-mikula-sonarsource** [comment]
Internal ticket [NET-1710](https://sonarsource.atlassian.net/browse/NET-1710)

[NET-1710]: https://sonarsource.atlassian.net/browse/NET-1710?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

**Comment by pavel-mikula-sonarsource** [comment]
Moved
